Source/UI/BillingForm/View/BillingFormTextFieldView.swift

```diff
@@ -5,6 +5,37 @@ protocol PhoneNumberTextFieldDelegate: AnyObject {
     func phoneNumberIsUpdated(number: String, tag: Int)
 }
 
+class BorderView: UIView {
+
+    private var style: ElementBorderStyle?
+
+    /// shaper layer that draw edges and corners
+    var borderLayer = CAShapeLayer()
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        layer.addSublayer(borderLayer)
+    }
+
+    // Keep border layer size in sync with owning view
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        guard let style = style else { return }
+        borderLayer.frame = bounds
+        borderLayer.updateEdgesAndCorners(with: style)
```

Comment:
We could also use an optional layer to pass the responsibility of only doing it once to the view instead of the layer. Currently, the layer inside `updateEdgesAndCorners` doesn't actually update, it checks if it exists and creates if not.

We could make a few minor changes and move that reponsability to the view, where it may be better suited:

```swift
private var borderLayer: BorderLayer?

override func layoutSubviews() {
         super.layoutSubviews()
         guard let style = style else { return }
         if borderLayer == nil {
             createLayer()
         }
         borderLayer?.frame = bounds
}

private func createLayer() {
    let borderLayer = CAShapeLayer()
    borderLayer.updateEdgesAndCorners(with: style)
    layer.addSublayer(borderLayer)
    self.borderLayer = borderLayer
}
```

We would now no longer need an init method and the CAShapeLayer extension would no longer need this code:
~`guard self.path == nil else { return }`~

Reply:
Could you describe why its not working? We should be able to achieve the same outcome while being clear and better enforcing responsibilities of involved objects, so if you say its not possible, I would like to understand what the challenge is.